### PR TITLE
CallHierarchyService Plugin API #3765

### PR DESCRIPTION
**Pull request by vrubezhny**
Issue #3765

Signed-off-by: Victor Rubezhny <vrubezhny@redhat.com>

### CODE REVIEW

**Comment by AlexTugarev** [comment]
Hi @vrubezhny! 'Just discovered this PR using the initially proposed `calls` protocol extension. 

Recently I've updated the protocol proposal as [requested](https://github.com/Microsoft/vscode-languageserver-node/pull/420#issuecomment-440288747). Please have a look and align.

A dev release of the TS LS implementing the updated `callHierarchy` protocol is published as well. The update for Theia `callhierarchy` extension is in progress.

**Comment by vrubezhny** [comment]
Hi @AlexTugarev ! 

I'll take a look/ Thank you to let me notest!

**Comment by tsmaeder** [comment]
@benoitf @evidolob @AlexTugarev ping?

**Comment by AlexTugarev** [comment]
https://github.com/microsoft/vscode-languageserver-node/pull/420 was accepted, so it would make sense to update. same applies to https://github.com/theia-ide/theia/pull/4154 btw 😞

File: packages/plugin/src/theia.d.ts [hunk 1]
```diff
@@ -6209,6 +6209,19 @@ declare module '@theia/plugin' {
...
+         *
+         * Multiple provider can be registered for a language. In that case providers are asked in
+         * parallel and the results are merged. A failing provider (rejected promise or exception) will
+         * not cause a failure of the whole operation.
+         *
+         * @param selector A selector that defines the documents this provider is applicable to.
+         * @param service A call hierarchy provider.
+         * @return A [disposable](#Disposable) that unregisters this provider when being disposed.
+         */
+        export function registerCallHierarchyProvider(selector: DocumentSelector, provider: CallHierarchyProvider): Disposable;
```
Comment (akosyakov) [inline packages/plugin/src/theia.d.ts]: Is it proposed api? Please move it theia.proposed.d.ts

Comment (akosyakov) [reply packages/plugin/src/theia.d.ts]: We probably should copy it from here: https://github.com/microsoft/vscode-languageserver-node/blob/51ce5233b6f97a4e2b6bee8f729205774a243140/client/src/callHierarchy.proposed.ts#L185-L234

Please don't forget a CQ. ping @theia-ide/ip-help if you need any help with it

**Pull request closed without merging**